transfer_client_portfolio/wizard/transfer_portfolio_wizard.py

```diff
@@ -0,0 +1,296 @@
+                    self._get_vals_transfer_registry(
+                        {
+                            "previous_salesperson_id": rbs,
+                            "list_opportunity_ids": records_by_salesperson[rbs]["opt"],
+                            "list_contacts_ids": records_by_salesperson[rbs][
+                                "contacts"
+                            ],
+                            "list_activity_ids": records_by_salesperson[rbs][
+                                "activities"
+                            ],
+                        }
+                    )
+                )
+                self.env["crm.lead"].browse(records_by_salesperson[rbs]["opt"]).write(
+                    {"previous_salesperson_id": rbs, "user_id": self.new_salesperson}
+                )
+                self.env["res.partner"].browse(
+                    records_by_salesperson[rbs]["contacts"]
+                ).write(
+                    {"previous_salesperson_id": rbs, "user_id": self.new_salesperson}
+                )
+                self.env["mail.activity"].browse(
+                    records_by_salesperson[rbs]["activities"]
+                ).write(
+                    {"previous_salesperson_id": rbs, "user_id": self.new_salesperson}
+                )
+            self.env["portfolio.transfer.registry"].create(ptr_vals)
+        else:
+            for sel in transfer_ids:
+                vals = sel._get_vals_transfer_registry()
+                sel.env["portfolio.transfer.registry"].create(vals)
+                self.env["res.partner"].browse(sel.contact_ids.ids).write(
+                    {
+                        "previous_salesperson_id": sel.current_salesperson.id,
+                        "user_id": sel.new_salesperson.id,
+                    }
+                )
+                self.env["crm.lead"].browse(sel.opportunity_ids.ids).write(
+                    {
+                        "previous_salesperson_id": sel.current_salesperson.id,
+                        "user_id": sel.new_salesperson.id,
+                    }
+                )
+                self.env["mail.activity"].browse(sel.activity_ids.ids).write(
+                    {
+                        "previous_salesperson_id": sel.current_salesperson.id,
+                        "user_id": sel.new_salesperson.id,
+                    }
+                )
+
+    def clear_records(self):
+        self.write(
+            {"contact_ids": False, "opportunity_ids": False, "activity_ids": False}
+        )
+        return {
+            "name": "Transfer Portfolio",
+            "view_mode": "form",
+            "view_id": False,
+            "res_model": self._name,
+            "domain": [],
+            "context": dict(self._context, active_ids=self.ids),
+            "type": "ir.actions.act_window",
+            "target": "new",
+            "res_id": self.id,
+        }
+
+    def _get_vals_transfer_registry(self, vals_def=None):
```

Comment:
I think if vals_def = {} or vals_def is defined as **vals_def the following if statement can be avoided.

Reply:
Fixed